tests/cache/models/subjects/getAll.js

```diff
@@ -0,0 +1,279 @@
+      expect(res.body[2].name).to.be.equal(subject3.substr(subject1.length + 1));
+      done();
+    });
+  });
+
+  it('fields filter returns expected number of keys', (done) => {
+    api.get(`${path}?fields=name`)
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+
+      expect(res.body.length).to.be.equal(3);
+      expect(Object.keys(res.body[0]).length).to.equal(3);
+      expect(Object.keys(res.body[1])).to.contain('id', 'name', 'apiLinks');
+      done();
+    });
+  });
+
+  it('fields filter returns apiLinks', (done) => {
+    api.get(`${path}?fields=name`)
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+
+      expect(res.body.length).to.be.equal(3);
+      expect(Array.isArray(res.body[0].apiLinks)).to.be.true;
+      expect(Array.isArray(res.body[1].apiLinks)).to.be.true;
+      expect(Array.isArray(res.body[2].apiLinks)).to.be.true;
+      done();
+    });
+  });
+
+  it('limit filter', (done) => {
+    api.get(`${path}?limit=1`)
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+
+      expect(res.body.length).to.be.equal(1);
+      expect(res.body[0].name).to.be.equal(subject1);
+      done();
+    });
+  });
+
+  it('offset filter', (done) => {
+    api.get(`${path}?offset=1`)
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+
+      expect(res.body.length).to.be.equal(2);
+      expect(res.body[0].absolutePath).to.be.equal(subject2);
+      expect(res.body[1].absolutePath).to.be.equal(subject3);
+      done();
+    });
+  });
+
+  it('name filter', (done) => {
+    api.get(`${path}?name=___Subject1`)
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+
+      expect(res.body.length).to.be.equal(1);
+      expect(res.body[0].name).to.be.equal(subject1);
+      expect(res.body[0].absolutePath).to.be.equal(subject1);
+      done();
+    });
+  });
+
+  it('combined filters', (done) => {
+    const filterstr = 'limit=2&offset=1&name=___*&' +
+    'sort=-name&fields=name,absolutePath';
+    api.get(`${path}?${filterstr}`)
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+
+      expect(res.body.length).to.be.equal(2);
+      expect(res.body[0].absolutePath).to.be.equal(subject2);
+      expect(res.body[1].absolutePath).to.be.equal(subject1);
+      expect(Object.keys(res.body[1])).to.contain('name', 'absolutePath');
+      done();
+    });
+  });
+
+  it('trailing asterisk is treated as "starts with"', (done) => {
+    api.get(path + '?name=' + tu.namePrefix + '*')
```

Comment:
Can you please apply a filter which returns a subset and not all of them. Like? absolutePath = .....*